hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -168,7 +168,8 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       if (tbl != null) {
         // If we try to create the table but the metadata location is already set, then we had a concurrent commit
         if (base == null && tbl.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP) != null) {
-          throw new AlreadyExistsException("Table already exists: %s.%s", database, tableName);
+          commitFailed = true;
+          throw new CommitFailedException("Table already exists: %s.%s", database, tableName);
```

Comment:
As noted in SnapshotProducer, we must throw a commit failed exception if we expect cleanup